### PR TITLE
[usage] More useful notification

### LINKED ISSUE #13166: Improve content of notifications when the Usage Limit is reached
The current notification text does not provide enough guidance.

To fix:

- Follow the same detailed explanation as the workspace start error message which mentions the team 
- Both should also link to your personal billing settings page where you can change billing accounts.

Existing notifications:
Screenshot 2022-09-20 at 18 47 33

Screenshot 2022-09-19 at 16 22 09

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Shows: 
1. The team/user attributed in the billing. 
2. The link to the relevant billing page.

| REACHED | 80% |
| - | - |
| Screenshot 2022-09-29 at 20 36 00 | Screenshot 2022-09-30 at 12 47 09 |

## Related Issue(s)

Fixes #13166

## How to test
[Preview env](https://lau-notifi6e179161fd.preview.gitpod-dev.com/)
1. Join [this team](https://lau-notifi6e179161fd.preview.gitpod-dev.com/teams/join?inviteId=85c97e27-4e1e-4b49-85c9-ec4f0c14fbe4).
2. You should see the notification already.

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-payment
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by jldec** [comment]
And for the personal usage account. 

> You have reached your usage limit.

**Comment by laushinka** [comment]
@jldec Changes done 🚀

**Comment by easyCZ** [comment]
Would it make sense to include in the warning message that 80% has been reached? Right now it says almost reached, but doesn't specify how urgent or how close the user is.
[See internal thread](https://gitpod.slack.com/archives/C02EN94AEPL/p1664477661094859)

**Comment by laushinka** [comment]
> Would it make sense to include in the warning message that 80% has been reached? [See internal thread](https://gitpod.slack.com/archives/C02EN94AEPL/p1664477661094859)

@easyCZ Thanks! I can see how the uncertainty of the word "almost" can create some panic. If we specify a number with a wording like "Your team has reached 80% of its usage limit", we would have to commit reflecting the actual percentage, won't we? For example today I see the banner, but then I keep racking up the usage without increasing the limit, tomorrow or the next  it wouldn't be 80% anymore.

**Comment by easyCZ** [comment]
We could also include the actual percentage in the message. We already check for 0.8 * usage limit so we know how much it is.

File: components/dashboard/src/AppNotifications.tsx [hunk 1]
```diff
@@ -54,7 +55,18 @@ export function AppNotifications() {
                 showIcon={true}
                 className="flex rounded mb-2 w-full"
             >
-                <span>{topNotification}</span>
+                <span>{topNotification}</span> Manage
+                <a
+                    className="gp-link hover:text-gray-600"
+                    href={
+                        notifications.length > 1
+                            ? `${gitpodHostUrl}t/${notifications[notifications.length - 1]}/billing`
```
Comment (easyCZ) [inline components/dashboard/src/AppNotifications.tsx:63]: Is it possible to use the `topNotification` variable here? It would avoid us having to do the lookup.

Comment (easyCZ) [reply components/dashboard/src/AppNotifications.tsx:63]: Do we actually need the `gitpodHostURL`? I think we could make the URL relative which would avoid having to specify it here

File: components/server/ee/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -2254,10 +2254,28 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
...
+                        if (teamOrUser) {
+                            result.push(teamOrUser?.slug);
+                            if (limit.reached) {
+                                result.unshift(`Your team '${teamOrUser?.name}' has reached its usage limit.`);
+                            } else if (limit.almostReached) {
+                                result.unshift(`Your team '${teamOrUser?.name}' has almost reached its usage limit.`);
+                            }
+                        }
+                        break;
+                    }
```
Comment (easyCZ) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:2278]: The nesting here makes it quite hard to follow the main flow of the method. What if we refactored this block into a separate method and do something like 
```
const usageNotification = getUsageNotificationForAttributionID(...)
if usageNotification.length > 0 { ... }
```

File: components/dashboard/src/AppNotifications.tsx (hunk 1 above)
Comment (laushinka) [reply components/dashboard/src/AppNotifications.tsx:63]: > Is it possible to use the `topNotification` variable here?

How do you mean? `topNotification` is the first sentence that includes the team name. The second entry in the array is the `Manage billing` link.

Comment (easyCZ) [reply components/dashboard/src/AppNotifications.tsx:63]: Ah, my bad. I misread the contents of those.

**Comment by laushinka** [comment]
@jldec also agrees with putting 80% so I will replace the `almost` sentences with "... has reached 80% of its usage limit.'

**Comment by easyCZ** [comment]
I think you should say `80% or more` as it starts showing at 80%, but continues to show even above that

**Comment by laushinka** [comment]
> I think you should say `80% or more` as it starts showing at 80%, but continues to show even above that

Yes, nice suggestion!

**Comment by jldec** [comment]
@laushinka I just tested this in a [preview env ](https://lau-notifi6e179161fd.preview.gitpod-dev.com/workspaces). 
After adding a team and configuring a test payment method for UBP I was immediately shown this notification.
 
Screenshot 2022-09-30 at 10 43 06

**Comment by laushinka** [comment]
> @laushinka I just tested this in a [preview env ](https://lau-notifi6e179161fd.preview.gitpod-dev.com/workspaces). After adding a team and configuring a test payment method for UBP I was immediately shown this notification.

Aha, thanks!

**Review by jldec — APPROVED** [review]
LGTM
Screenshot 2022-09-30 at 13 08 35

File: components/server/ee/src/workspace/gitpod-server-impl.ts (hunk 2 above)
Comment (laushinka) [reply components/server/ee/src/workspace/gitpod-server-impl.ts]: Sorry @easyCZ I was refactoring this but forgot to hold and now it's merged!
